cassandra/cqlengine/models.py

```diff
@@ -705,18 +705,23 @@ def update(self, **values):
         It is possible to do a blind update, that is, to update a field without having first selected the object out of the database.
         See :ref:`Blind Updates <blind_updates>`
         """
-        for k, v in values.items():
-            col = self._columns.get(k)
+        for column_id, v in values.items():
+            col = self._columns.get(column_id)
 
             # check for nonexistant columns
             if col is None:
-                raise ValidationError("{0}.{1} has no column named: {2}".format(self.__module__, self.__class__.__name__, k))
+                raise ValidationError(
+                    "{0}.{1} has no column named: {2}".format(
+                        self.__module__, self.__class__.__name__, column_id))
 
-            # check for primary key update attempts
-            if col.is_primary_key:
-                raise ValidationError("Cannot apply update to primary key '{0}' for {1}.{2}".format(k, self.__module__, self.__class__.__name__))
+            setattr(self, column_id, v)
```

Comment:
Thanks @mambocab for the feedback! I may have overlooked something here. I'll double check that soon.